elements/userlevel/todpdkdevice.cc

```diff
@@ -124,7 +124,16 @@ inline struct rte_mbuf* get_mbuf(Packet* p, bool create=true) {
 
     if (likely(DPDKDevice::is_dpdk_packet(p))) {
         mbuf = (struct rte_mbuf *) p->destructor_argument();
-        p->set_buffer_destructor(DPDKDevice::fake_free_pkt);
+        rte_pktmbuf_pkt_len(mbuf) = p->length();
```

Comment:
Shouldn't you use p->buffer_length() here and below? User might have used pull() or take() on this packet.